sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -579,6 +581,18 @@ class ResolveSessionCatalog(val catalogManager: CatalogManager)
   }
 
   object ResolvedV1Identifier {
+    def unapply(resolved: LogicalPlan): Option[TableIdentifier] = resolved match {
+      case ResolvedIdentifier(catalog, ident) if supportsV1Command(catalog) =>
```

Comment:
@amaliujia @cloud-fan 
This change looks to have broken creating V1 table from a `V2_SESSION_CATALOG_IMPLEMENTATION` like [Iceberg's SparkSessionCatalog](https://iceberg.apache.org/docs/nightly/spark-configuration/#replacing-the-session-catalog)

https://github.com/apache/spark/blob/e693e18c7d0e9a495dcb6e9b31dac9ce2b98428c/sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala#L159-L170

Reply:
Does Iceberg catalog extend `DelegatingCatalogExtension`?

Reply:
We do want to use v2 commands for custom catalogs that do not extend `DelegatingCatalogExtension`

Reply:
> Does Iceberg catalog extend DelegatingCatalogExtension?
Nope.

> We do want to use v2 commands for custom catalogs that do not extend DelegatingCatalogExtension
Even so, is it the right time to introduce such a behavior change in a bug fix release?

Reply:
We can consider it as a bug. People implementing DS V2 catalog APIs expect to see v2 commands to customize the table behaviors. And there is a backdoor: `DelegatingCatalogExtension`.

For iceberg, it should be easy to work around it by extending `DelegatingCatalogExtension`? Iceberg catalog can still keep all its methods unchanged, don't use the `delegate`.

Reply:
Iceberg catalog already extends a base class so it can not extend `DelegatingCatalogExtension`.

Reply:
Iceberg's [SparkSessionCatalog](https://github.com/apache/iceberg/blob/main/spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java) already extends a base class. There's no easy way to extend `DelegatingCatalogExtension` without a major refactoring.

Reply:
We need to make either the iceberg `BaseCatalog` or the Spark `DelegatingCatalogExtension` an interface. It looks easier to make `BaseCatalog` an interface?